test/src/parse/test_replace.cpp

```diff
@@ -0,0 +1,165 @@
+
+#include <gtest/gtest.h>
+
+#include <parse/parselo.h>
+
+#include "globalincs/safe_strings.h"
+
+#include "util/FSTestFixture.h"
+
+class ReplaceTest : public test::FSTestFixture {
```

Comment:
Thanks for taking the time to write tests! :+1:

Reply:
Optional: it might be good if the test for the `replace_one()` overload that uses C-strings also tested for return value of -3. The tests you have already cover a lot of ground, though.